docs/cookbook/getInstantOfNearestOffsetTransitionToInstant.mjs

```diff
@@ -0,0 +1,34 @@
+/**
+ * Get the nearest following instant that the given time zone transitions to
+ * another UTC offset, inclusive or exclusive.
+ *
+ * @param {Temporal.Absolute} absolute - Start time to consider
+ * @param {Temporal.TimeZone} timeZone - Time zone to consider
+ * @param {boolean} inclusive - Include the start time, or not
+ * @returns {(Temporal.Absolute|null)} - Next UTC offset transition, or null if
+ *   none known at this time
+ */
+function getInstantOfNearestOffsetTransitionToInstant(absolute, timeZone, inclusive) {
+  let nearest;
+  if (inclusive) {
+    // In case absolute itself is the moment of a transition:
+    [nearest] = timeZone.getTransitions(absolute.minus({ nanoseconds: 1 }));
+  } else {
+    [nearest] = timeZone.getTransitions(absolute);
+  }
+  return nearest;
+}
+
+const absolute = Temporal.Absolute.from('2020-04-16T21:01Z');
+
+const nyc = Temporal.TimeZone.from('America/New_York');
+const nextTransition = getInstantOfNearestOffsetTransitionToInstant(absolute, nyc, false);
+assert.equal(nextTransition.toString(), '2020-11-01T06:00Z');
```

Comment:
Nit: prefer not to use dates in the future... Who knows if NYC will still exist in November?